### ISSUE TITLE
nacos autorefresh feature not working with jasypt

### ISSUE THREAD

**Issue by benhouse1987** [ulisesbocchio/jasypt-spring-boot#186]
```		<dependency>
			<groupId>com.github.ulisesbocchio</groupId>
			<artifactId>jasypt-spring-boot-starter</artifactId>
			<version>3.0.2</version>
		</dependency>
               <dependency>
			<groupId>com.alibaba.boot</groupId>
			<artifactId>nacos-config-spring-boot-starter</artifactId>
			<version>0.2.4</version>
		</dependency>
```
```   
@NacosValue(value = "${test:}",autoRefreshed = true)
````

Nacos autoRefreshed flag not working when import jasypt in pom.

guess there are some caching problems.

hope fix

**Comment by benhouse1987** [comment]
hope fix soon

**Comment by Comven** [comment]
I had the same problem, The version 2.1.2 with algorithm of PBEWithMD5AndDES was  OK，version 3.0.2 with algorithm of  PBEWITHHMACSHA512ANDAES_256 Not working properly. I don't know if it only appears when using Spring-Cloud-Alibaba

**Comment by AlexanderLuo** [comment]
CachingDelegateEncryptablePropertySource#getProperty      The cache! ! ! !!

**Comment by Comven** [comment]
Is the problem fixed，did you resolved it @benhouse1987

**Comment by jianghao1123** [comment]
> Is the problem fixed，did you resolved it @benhouse1987

try
`
@Component
public class EncryptablePropertySourceCacheRefresher {

    private final ConfigurableEnvironment environment;

    public EncryptablePropertySourceCacheRefresher(ConfigurableEnvironment environment) {
        this.environment = environment;
    }

    private void refreshCachedProperties() {
        PropertySources propertySources = environment.getPropertySources();
        propertySources.forEach(this::refreshPropertySource);
    }

    @SuppressWarnings("rawtypes")
    private void refreshPropertySource(PropertySource<?> propertySource) {
        if (propertySource instanceof CompositePropertySource) {
            CompositePropertySource cps = (CompositePropertySource) propertySource;
            cps.getPropertySources().forEach(this::refreshPropertySource);
        } else if (propertySource instanceof EncryptablePropertySource) {
            EncryptablePropertySource eps = (EncryptablePropertySource) propertySource;
            eps.refresh();
        }
    }

    @EventListener(NacosConfigReceivedEvent.class)
    public void onApplicationEvent(NacosConfigReceivedEvent event) {
        refreshCachedProperties();
    }

    @EventListener(ApplicationReadyEvent.class)
    public void onApplicationEvent(ApplicationReadyEvent event) {
        refreshCachedProperties();
    }
}

`

**Comment by Comven** [comment]
> > Is the problem fixed，did you resolved it @benhouse1987
> 
> try
> `
> @component
> public class EncryptablePropertySourceCacheRefresher {
> 
> ```
> private final ConfigurableEnvironment environment;
> 
> public EncryptablePropertySourceCacheRefresher(ConfigurableEnvironment environment) {
>     this.environment = environment;
> }
> 
> private void refreshCachedProperties() {
>     PropertySources propertySources = environment.getPropertySources();
>     propertySources.forEach(this::refreshPropertySource);
> }
> 
> @SuppressWarnings("rawtypes")
> private void refreshPropertySource(PropertySource<?> propertySource) {
>     if (propertySource instanceof CompositePropertySource) {
>         CompositePropertySource cps = (CompositePropertySource) propertySource;
>         cps.getPropertySources().forEach(this::refreshPropertySource);
>     } else if (propertySource instanceof EncryptablePropertySource) {
>         EncryptablePropertySource eps = (EncryptablePropertySource) propertySource;
>         eps.refresh();
>     }
> }
> 
> @EventListener(NacosConfigReceivedEvent.class)
> public void onApplicationEvent(NacosConfigReceivedEvent event) {
>     refreshCachedProperties();
> }
> 
> @EventListener(ApplicationReadyEvent.class)
> public void onApplicationEvent(ApplicationReadyEvent event) {
>     refreshCachedProperties();
> }
> ```
> 
> }
> 
> `

Thinks, i find it fixed with version 3.0.3

**Comment by ulisesbocchio** [comment]
resolved in 3.0.3 with improved refreshed events

**Comment by zhang6332** [comment]
spring-boot2.6.7/spring-cloud-alibaba2021.0.1.0/jasypt-spring-boot-starter3.0.3 ,Nacos autoRefreshed flag not working when (jasypt.encryptor.password= jasypt.encryptor.algorithm=) jasypt was writte in properties file
